analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeHelperTest.cs

```diff
@@ -89,5 +89,412 @@ public void Type_GetSymbolType_Alias()
+        }
+
+        [DataTestMethod]
+        [DataRow("object")]
+        [DataRow("System.IComparable")]
+        [DataRow("int?")]
+        [DataRow("System.Nullable<int>")]
+        public void IsNonNullableValueType_False_Simple(string type)
+        {
+            var fieldSymbol = FirstFieldSymbolFromCode($$"""
+                class Test
+                {
+                    {{type}} field;
+                }
+                """);
+            fieldSymbol.Type.IsNonNullableValueType().Should().BeFalse();
+        }
+
+        [DataTestMethod]
+        [DataRow("struct")]
+        [DataRow("unmanaged")]
+        public void IsNonNullableValueType_Generic(string typeConstraint)
+        {
+            var fieldSymbol = FirstFieldSymbolFromCode($$"""
+                using System;
+                class Test<T> where T: {{typeConstraint}}
+                {
+                    T field;
+                }
+                """);
+            fieldSymbol.Type.IsNonNullableValueType().Should().BeTrue();
+        }
+
+        [DataTestMethod]
+        [DataRow("T?")]
+        [DataRow("Nullable<T>")]
+        public void IsNonNullableValueType_Generic_ConstraintStruct_Nullable(string type)
+        {
+            var fieldSymbol = FirstFieldSymbolFromCode($$"""
+                using System;
+                class Test<T> where T: struct
+                {
+                    {{type}} field;
+                }
+                """);
+            fieldSymbol.Type.IsNonNullableValueType().Should().BeFalse();
+        }
+
+        [DataTestMethod]
+        [DataRow("")]                      // Unbounded (can be reference type or value type)
+        [DataRow("where T: new()")]        // Unbounded
+        [DataRow("where T: notnull")]      // Unbounded
+        [DataRow("where T: struct")]
+        [DataRow("where T: unmanaged")]
+        [DataRow("where T: Enum")]
+        [DataRow("where T: struct, Enum")]
+        [DataRow("where T: struct, Enum, IComparable")]
+        [DataRow("where T: class")]
+        [DataRow("where T: class?")]
+        [DataRow("where T: class, new()")]
+        [DataRow("where T: Exception")]
+        [DataRow("where T: Exception?")]
+        [DataRow("where T: IComparable")]
+        [DataRow("where T: IComparable?")]
+        [DataRow("where T: Delegate")]
+        [DataRow("where T: Delegate?")]
+        public void IsNonNullableValueType_Generic_Constraint_Nullable(string constraint)
+        {
+            var fieldSymbol = FirstFieldSymbolFromCode($$"""
+                #nullable enable
+                using System;
+                class Test<T> {{constraint}}
+                {
+                    T? field;
+                }
+                """);
+            fieldSymbol.Type.IsNonNullableValueType().Should().BeFalse();
+        }
+
+        [DataTestMethod]
+        [DataRow("int?")]
+        [DataRow("System.Nullable<int>")]
+        public void IsNullableValueType_Simple(string type)
```

Comment:
Same as the `Non` version
```suggestion
        public void IsNullableValueType_Simple_True(string type)
```